src/components/data-exploration/hierarchical-tree/HierarchicalTree.jsx

```diff
@@ -50,24 +50,27 @@ const HierarchicalTree = (props) => {
     // If rootNode, ignore
     if (dragNode.rootNode) return;
 
+    if (dropPosition === -1) return;
+
     // pos is a string e.g.: 0-0-1, each number is a position in a tree level
     const posFromArray = dragNode.pos.split('-');
     const posToArray = node.pos.split('-');
 
-    const fromPosition = parseInt(posFromArray[2], 10);
+    const sameLevel = (posFromArray.length === posToArray.length);
 
     // If not in the same cellClass, ignore
     if (!_.isEqual(posFromArray[1], posToArray[1])) return;
 
-    // If was dropped in same place, ignore
-    if (fromPosition === dropPosition) return;
+    const numberOfClusters = treeData[posFromArray[1]].children.length;
 
-    // If not dropped in gap, ignore
-    // (only allow dropToGap when the destination node is rootNode
-    // because it can have children nodes)
-    if (!dropToGap && !node.rootNode) return;
+    const fromPosition = parseInt(posFromArray[2], 10);
+    const toPosition = (!sameLevel ? dropToGap ? numberOfClusters : 0 : dropPosition);
+
+    // If was dropped in same place, ignore
+    if (fromPosition === toPosition) return;
 
-    const newPosition = dropPosition - (fromPosition < dropPosition ? 1 : 0);
+    // if dropping below the initial position subtract 1, if dropping to secondary position add 1
+    const newPosition = toPosition - (fromPosition < toPosition ? 1 : 0) + (!sameLevel ? 0 : node.dragOver);
```

Comment:
What is `node.dragOver`? Is it's value an integer?

Reply:
I see that the value of `dragOver` is a boolean. Although it is represented as an integer, I think it is much better to have a switch (e.g. `addDragOverPosition = node.dragOver ? 1 : 0`) rather than directly using arithmetic operations on it.